### ISSUE TITLE
some issues for incremental-learning

### ISSUE THREAD

**Issue by wangjl1993** [vcl-iisc/DeGAN#1]
Hi~ thank you for the provided code, its cool. I have some issues:
1. when we apply DeGan for incremental-learning, for incremental model, we have fake img (old task data) and new task data (real img). For real img, DeGan used one-hot cross-entropy loss for all classifier. For fake img, DeGan used distill loss for old classifier ? (then, what about new classifier?)
Can you provided more details on incremental-learning experiment? thanks. 

2. In train_teacher.py, tfm = transforms.Compose([ transforms.ToTensor()]); 
 In dfgan.py     dataset = dset.CIFAR100(root=opt.dataroot, download=True,
                               transform=transforms.Compose([
                               transforms.Scale(opt.imageSize),
                               transforms.ToTensor(),
                               transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)),
                           ]))
Is the Normalize irrelevant?

3. Can I use a small batch size for training dfgan.py?

**Comment by hongge831** [comment]
@yixiaonaihe hi have you figure out the problems you mentioned? I also have the same question on the class incremental learning task.

**Comment by SravantiAddepalli** [comment]
@yixiaonaihe @hongge831 
Thank you for your interest in our work. Apologies for the delayed response.

1. In the Incremental Learning setting, we assume that we have a classifier trained on old classes. For all further experiments, weights of this pre-trained classifier are frozen. Although this pre-trained classifier is available to us, the old class data is not available. Only the new class data is assumed to be available. We use DeGAN framework to generate representative samples for the old classes, using new class data as the Proxy dataset. 
For incremental learning, we train a new classifier and use cross-entropy loss for the new classes (on True dataset) and distillation loss on the generated data (and Proxy data) to avoid catastrophic forgetting of old classes. The distillation loss ensures that for the generated data (and Proxy data), output of the new network matches with that of the old classifier. Therefore, distillation loss is used to train the new classifier (not the old one). 
In addition, we also add a regularization term to account for the relative scaling of logits between the old and new classes.

2. The normalization is used to shift the range of the image from [0,1] to [-1,1]. This ensures that the images used to train the teacher are in the same range as the proxy data generated using the generator (which has tanh as the final activation). 

3. It is recommended to use a large batch size because diversity loss ensures class balance among each batch of data samples. So, larger batch size would ensure a better class balance in the generated data. However, batch size of 512 and above would also work for CIFAR-100. For CIFAR-10, batch size can be above 128.
